Can't disable memcached session support

I tried to configure and then make php-memcached without session support and it fails. 

```
**./configure --enable-memcached-igbinary --disable-memcached-sasl --disable-memcached-session**
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.16 (ok)
checking for gawk... gawk
checking whether to enable memcached support... yes, shared
checking for libmemcached... yes, shared
checking whether to enable memcached session handler support... no
checking whether to enable memcached igbinary serializer support... yes
checking whether to enable memcached json serializer support... no
checking whether to enable memcached msgpack serializer support... no
checking whether to enable memcached sasl support... no
checking whether to enable memcached protocol support... no
checking whether to use system FastLZ library... no
checking for ZLIB... yes, shared
checking for pkg-config... /usr/bin/pkg-config
checking for zlib location... /usr
checking for igbinary includes... /usr/include/php/20151012
checking for memcached session support... disabled
checking for memcached igbinary support... enabled
checking for memcached msgpack support... disabled
checking for libmemcached location... found version 1.0.18, under /usr
checking for libmemcached sasl.h requirement... yes
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking sasl/sasl.h usability... yes
checking sasl/sasl.h presence... yes
checking for sasl/sasl.h... yes
checking whether to enable sasl support... no
checking whether memcached_exist is defined... yes
checking whether memcached_set_encoding_key is defined... yes
checking for memcached protocol support... disabled
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing libtool commands

**root@FPM1:/usr/local/src/php-memcached# make**
/bin/bash /usr/local/src/php-memcached/libtool --mode=compile cc  -I/usr/include/php/20151012   -I. -I/usr/local/src/php-memcached -DPHP_ATOM_INC -I/usr/local/src/php-memcached/include -I/usr/local/src/php-memcached/main -I/usr/local/src/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib   -c /usr/local/src/php-memcached/php_memcached.c -o php_memcached.lo 
libtool: compile:  cc -I/usr/include/php/20151012 -I. -I/usr/local/src/php-memcached -DPHP_ATOM_INC -I/usr/local/src/php-memcached/include -I/usr/local/src/php-memcached/main -I/usr/local/src/php-memcached -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -DHAVE_CONFIG_H -g -O2 -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -c /usr/local/src/php-memcached/php_memcached.c  -fPIC -DPIC -o .libs/php_memcached.o
In file included from /usr/local/src/php-memcached/php_memcached.c:25:0:
/usr/local/src/php-memcached/php_memcached.c: In function ‘OnUpdateConsistentHash’:
/usr/local/src/php-memcached/php_memcached_private.h:220:50: error: ‘zend_php_memcached_globals {aka struct _zend_php_memcached_globals}’ has no member named ‘session’
 #  define MEMC_SESS_INI(v) (php_memcached_globals.session.v)
                                                  ^
/usr/local/src/php-memcached/php_memcached.c:330:3: note: in expansion of macro ‘MEMC_SESS_INI’
   MEMC_SESS_INI(consistent_hash_type) = MEMCACHED_BEHAVIOR_KETAMA;
   ^
/usr/local/src/php-memcached/php_memcached_private.h:220:50: error: ‘zend_php_memcached_globals {aka struct _zend_php_memcached_globals}’ has no member named ‘session’
 #  define MEMC_SESS_INI(v) (php_memcached_globals.session.v)
                                                  ^
/usr/local/src/php-memcached/php_memcached.c:332:3: note: in expansion of macro ‘MEMC_SESS_INI’
   MEMC_SESS_INI(consistent_hash_type) = MEMCACHED_BEHAVIOR_KETAMA;
   ^
/usr/local/src/php-memcached/php_memcached_private.h:220:50: error: ‘zend_php_memcached_globals {aka struct _zend_php_memcached_globals}’ has no member named ‘session’
 #  define MEMC_SESS_INI(v) (php_memcached_globals.session.v)
                                                  ^
/usr/local/src/php-memcached/php_memcached.c:334:3: note: in expansion of macro ‘MEMC_SESS_INI’
   MEMC_SESS_INI(consistent_hash_type) = MEMCACHED_BEHAVIOR_KETAMA_WEIGHTED;
   ^
Makefile:194: recipe for target 'php_memcached.lo' failed
make: *** [php_memcached.lo] Error 1
```

_[edited to add code-block backticks]_

--- a/php_memcached.c
+++ b/php_memcached.c
@@ -302,6 +302,7 @@ static PHP_INI_MH(OnUpdateSerializer)
 	return OnUpdateString(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
 }
 
+#ifdef HAVE_MEMCACHED_SESSION
 static
 PHP_INI_MH(OnUpdateDeprecatedLockValue)
 {
@@ -342,6 +343,7 @@ PHP_INI_MH(OnUpdateConsistentHash)
 	}
 	return OnUpdateString(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
 }
+#endif // HAVE_MEMCACHED_SESSION
 
 #define MEMC_INI_ENTRY(key, default_value, update_fn, gkey) \
 	STD_PHP_INI_ENTRY("memcached."key, default_value, PHP_INI_ALL, update_fn, memc.gkey, zend_php_memcached_globals, php_memcached_globals)

